### ISSUE TITLE
error in function sjSDM.tune

### ISSUE THREAD

**Issue by mtva0001** [TheoreticalEcology/s-jSDM#88]
Hi,

Thanks for doing some improvements and updates! I just tried this new function but it seems something is not working properly. Could you please check this?

**This is what I try to run:**
tune2 = sjSDM_cv(env = scale(as.matrix(Env)), Y = as.matrix(Occ), 
                learning_rate = 0.01, iter = 100L, CV = 5,
                tune_steps = 40,
                lambda_cov = seq(0, 0.1, 0.01),
                lambda_coef = seq(0, 0.1, 0.01),
                alpha_cov = seq(0, 1, 0.05),
                alpha_coef = seq(0, 1, 0.05),
                alpha_spatial = seq(0, 1, 0.05),
                sampling = 100L,
                biotic =  bioticStruct(df=dim(Occ)[2]),
                spatial = linear(scale(SPV), ~0+.),
                lambda_spatial = 2^seq(-10, -0.5, length.out = 20),
                step_size = 5L,
                blocks = 3L,
                family=binomial("probit"),
                n_cores = 8)
#Done, tune2 object was plotted.

model2=sjSDM.tune(tune2)

**Message:**
> model2=sjSDM.tune(tune2)
Error in (function (Y = NULL, env = NULL, biotic = bioticStruct(), spatial = NULL,  : 
  unused arguments (family.family = "binomial", family.link = "probit", family.linkfun = function (mu) 
qnorm(mu), family.linkinv = function (eta) 
{
    thresh <- -qnorm(.Machine$double.eps)
    eta <- pmin(pmax(eta, -thresh), thresh)
    pnorm(eta)
}, family.variance = function (mu) 
mu * (1 - mu), family.dev.resids = function (y, mu, wt) 
.Call(C_binomial_dev_resids, y, mu, wt), family.aic = function (y, n, mu, wt, dev) 
{
    m <- if (any(n > 1)) n else wt
    -2 * sum(ifelse(m > 0, (wt/m), 0) * dbinom(round(m * y), round(m), mu, log = TRUE))
}, family.mu.eta = function (eta) 
pmax(dnorm(eta), .Machine$double.eps), family.initialize = {
    if (NCOL(y) == 1) {
        if (is.factor(y)) y <- y != levels(y)[1]
        n <- rep.int(1, nobs)
        y[weights == 0] <- 0
        if (any(y < 0 | y > 1)) stop("y values must be 0 <= y <= 1")
        mustart <- (weights * y + 0.5)/(weights + 1)
        m <- weights * y
        if ("binomial" == "binomial" && any(abs(m - round(m)) >

Many thanks!
Best,

Máté

**Comment by MaximilianPi** [comment]
Hi Máté,

thanks for reporting!

I found the issue, the current implementation accidentally (not sure what I have done there...) unlists list (e.g. family) objects:

https://github.com/TheoreticalEcology/s-jSDM/blob/08eb0ee5c3dc3f4e1eee6b6ba0fed8ae0635e199/sjSDM/R/sjSDM_cv.R#L284 

I pushed a fix into the development branch, you can install it via:

`devtools::install_github("https://github.com/TheoreticalEcology/s-jSDM", subdir = "sjSDM", ref="devel")`

Or just exclude 'family = binomial("probit")' which is the default anyway:

```
tune2 = sjSDM_cv(env = scale(as.matrix(Env)), Y = as.matrix(Occ),
learning_rate = 0.01, iter = 100L, CV = 5,
tune_steps = 40,
lambda_cov = seq(0, 0.1, 0.01),
lambda_coef = seq(0, 0.1, 0.01),
alpha_cov = seq(0, 1, 0.05),
alpha_coef = seq(0, 1, 0.05),
alpha_spatial = seq(0, 1, 0.05),
sampling = 100L,
biotic = bioticStruct(df=dim(Occ)[2]),
spatial = linear(scale(SPV), ~0+.),
lambda_spatial = 2^seq(-10, -0.5, length.out = 20),
step_size = 5L,
blocks = 3L,
#family=binomial("probit"),
n_cores = 8)
#Done, tune2 object was plotted.

model2=sjSDM.tune(tune2)
```

**Comment by mtva0001** [comment]
Thanks! It works now!

**Comment by MaximilianPi** [comment]
Let's keep it open for others with the same problem.
